backend/execution_cache/src/mutation.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ArgoWorkflowTemplate string = "workflows.argoproj.io/template"
+	ExecutionKey         string = "executionKey"
```

Comment:
We can add vendor prefix to the key:
`pipelines.kubeflow.org/execution_cache_key`

P.S. The cache key might be a bit better suited for labels (instead of annotations) as at some point you might use Kubernetes label filtering capabilities to query the running Pods.